src/app/InteractionModelEngine.cpp

```diff
@@ -374,9 +374,14 @@ bool InteractionModelEngine::SubjectHasPersistedSubscription(FabricIndex aFabric
     return false;
 }
 
+void InteractionModelEngine::OnDone(CommandResponseSender & apResponderObj)
+{
+    mCommandResponderObjs.ReleaseObject(&apResponderObj);
+}
+
 void InteractionModelEngine::OnDone(CommandHandler & apCommandObj)
 {
-    mCommandHandlerObjs.ReleaseObject(&apCommandObj);
+    VerifyOrDie(false);
```

Comment:
Needs documentation explaining why.

But also, feels like we should just fix our APIs so that this method doesn't need to exist here.  Followup for that, probably.